src/formats/airtable-api/attachment-helpers.ts

```diff
@@ -0,0 +1,224 @@
+/**
+ * Attachment handling for Airtable
+ */
+
+import { requestUrl, normalizePath, TFile } from 'obsidian';
+import type { Vault, App } from 'obsidian';
+import { ImportContext } from '../../main';
+import type { AirtableAttachment, AttachmentResult } from './types';
+import { sanitizeFileName } from '../../util';
+
+/**
+ * Download an attachment from Airtable
+ */
+export async function downloadAttachment(
+	attachment: AirtableAttachment,
+	context: {
+		ctx: ImportContext;
+		currentFolderPath: string;
+		currentFilePath: string;
+		vault: Vault;
+		app: App;
+		downloadAttachments: boolean;
+		getAvailableAttachmentPath: (filename: string) => Promise<string>;
+	}
+): Promise<AttachmentResult> {
+	const { ctx, vault, downloadAttachments, getAvailableAttachmentPath } = context;
+	
+	// If download is disabled, return URL
+	if (!downloadAttachments) {
+		return {
+			path: attachment.url,
+			isLocal: false,
+			filename: attachment.filename,
+		};
+	}
+	
+	try {
+		ctx.status(`Downloading attachment: ${attachment.filename}`);
+		
+		// Download the file
+		const response = await requestUrl({
+			url: attachment.url,
+			method: 'GET',
+			throw: false,
+		});
+		
+		if (response.status !== 200) {
+			console.warn(`Failed to download attachment: ${attachment.filename}`);
+			return {
+				path: attachment.url,
+				isLocal: false,
+				filename: attachment.filename,
+			};
+		}
+		
+		// Sanitize filename
+		const sanitized = sanitizeFileName(attachment.filename);
+		
+		// Get available path (respects user's attachment folder settings)
+		const targetPath = await getAvailableAttachmentPath(sanitized);
+		
+		// Create the file
+		const normalizedPath = normalizePath(targetPath);
+		await vault.createBinary(normalizedPath, response.arrayBuffer);
+		
+		// Return path without extension for wiki links
+		const pathWithoutExt = targetPath.replace(/\.[^/.]+$/, '');
+		
+		return {
+			path: pathWithoutExt,
```

Comment:
This doesn't feel like the right data to be returning because there are two instances below where the extension is extracted from the filename and then appended to the end of this pathWithoutExt.

Maybe just return the full path? Then the users of this data can call the various filepath utilities to get the exact format required.